### PR TITLE
C2C-261: Close stale visits 'scheduled task to run every hour by default'

### PR DESCRIPTION
**Pull request by kazlaw**
Link to ticket --> [here](https://mekomsolutions.atlassian.net/browse/C2C-261)

### CODE REVIEW

File: configuration/liquibase/liquibase.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                  http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+                  
+
+    <changeSet id="201905101200" author="Angshu,Bindu">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                select count(*) from scheduler_task_config where schedulable_class= "org.openmrs.module.emrapi.adt.CloseStaleVisitsTask" and name = "Close Stale Visits Task"; </sqlCheck>
+        </preConditions>
+        <comment> Add Bahmni preferred EMR API scheduler which closes stale visits </comment>
```
Comment (Ruhanga) [inline configuration/liquibase/liquibase.xml:12]:
```suggestion
        <comment> Modify Bahmni preferred EMR API scheduler to close stale visits every hour </comment>
```

**Review by Ruhanga — COMMENTED** [review]
LGTM just a little comment below. Have you tested this @kazlaw?

File: configuration/liquibase/liquibase.xml (hunk 1 above)
Comment (kazlaw) [reply configuration/liquibase/liquibase.xml:12]: Not yet @Ruhanga , will proceed with the test and let you know

**Review by rbuisson — COMMENTED** [review]
I don't think this is going to work. Does it?
This change set has already been processed by Bahmni Core, so I don't think it will re-run.
Additionally, because the comment has changed, it's possible that the changeset checksum will change too. Therefore throwing an error message upon processing.

Best would be to simply create a completely new change set, with the `changeSet id="201905101200` set to something new, so that it's processed like any new change.

**Comment by kazlaw** [comment]
> I don't think this is going to work. Does it? This change set has already been processed by Bahmni Core, so I don't think it will re-run. Additionally, because the comment has changed, it's possible that the changeset checksum will change too. Therefore throwing an error message upon processing.
> 
> Best would be to simply create a completely new change set, with the `changeSet id="201905101200` set to something new, so that it's processed like any new change.

Let me attempt this

**Review by Ruhanga — COMMENTED** [review]
Have you tried this out @kazlaw?

File: configuration/liquibase/liquibase.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    <changeSet id="202402271000" author="Kazooba,Lawrence">
+        <preConditions onFail="CONTINUE">
+            <sqlCheck expectedResult="1">
+                select count(*) from scheduler_task_config where schedulable_class= "org.openmrs.module.emrapi.adt.CloseStaleVisitsTask" and name = "Close Stale Visits Task"; </sqlCheck>
+        </preConditions>
+        <comment> Modify Bahmni preferred EMR API scheduler to close stale visits every hour </comment>
+        <update tableName="scheduler_task_config">
+                <column name="repeat_interval" value="3600"/>
+                <where>name="repeat_interval"</where>
```
Comment (rbuisson) [inline configuration/liquibase/liquibase.xml:15]: The where clause will evaluate a field on the record, not the field name itself.
Eg:
```sql
name = "Close Stale Visits Task";
```
or
```sql
uuid = xxx
```

In your case I would suggest to have the where clause to be the same as the `sqlCheck`:
```sql
<where>schedulable_class = "org.openmrs.module.emrapi.adt.CloseStaleVisitsTask" and name = "Close Stale Visits Task"</where>
```

**Review by rbuisson — CHANGES_REQUESTED** [review]

**Comment by kazlaw** [comment]
image

@rbuisson , @Ruhanga the changes from this commit [here](https://github.com/mekomsolutions/openmrs-config-c2c/pull/72/commits/33336975a01b30130f4dcd726bbd36bbfc651d11) insert the value as seen above.

**Review by rbuisson — APPROVED** [review]
@kazlaw , I don't think we need to worry about changing any other field. We should be good to go.

**Comment by kazlaw** [comment]
Noted, thanks @rbuisson

**Pull request merged**